### PR TITLE
support telechat2 model

### PR DESCRIPTION
**Pull request by shunxing12345**
## Motivation

support telechat2 model

## Modifications

support telechat2 model

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html).

### CODE REVIEW

File: python/sglang/srt/models/telechat2.py [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+
+# Adapted from llama2.py
+# Modify details for the adaptation of TeleChat2 model.
+"""Inference-only TeleChat2 model compatible with HuggingFace weights."""
+
+from typing import Any, Dict, Iterable, Optional, Tuple
+
+import torch
+from torch import nn
+from vllm.distributed import get_tensor_model_parallel_world_size
```
Comment (zhyncs) [inline python/sglang/srt/models/telechat2.py]: remove any vLLM components

**Review by zhyncs — CHANGES_REQUESTED** [review]
add unit test for this new model

File: python/sglang/srt/models/telechat2.py (hunk 1 above)
Comment (zhyncs) [reply python/sglang/srt/models/telechat2.py]: ref https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/models/llama.py

**Comment by zhyncs** [comment]
ref https://github.com/sgl-project/sglang/blob/main/test/srt/models/test_qwen_models.py

**Comment by shunxing12345** [comment]
> ref https://github.com/sgl-project/sglang/blob/main/test/srt/models/test_qwen_models.py

Thank you for your feedback! I will add the unit test for the new model as soon as possible. 😊

**Comment by shunxing12345** [comment]
fc03110d-83bc-4f54-aaa4-46cc43c0f7df
Hi @zhyncs,
I started working on it, but encountered the issue shown in the attached image. It seems to be related to the trust_remote_code setting when loading the configuration file. Could you please advise on the best way to resolve this? Thank you for your guidance!
